### DISCUSSION TITLE
Renicing things

### DISCUSSION THREAD

**Discussion by grahamperrin** [freebsd/poudriere#969] [Q&A]
### [nice(1)](https://www.freebsd.org/cgi/man.cgi?query=nice&sektion=1&manpath=FreeBSD)

https://github.com/freebsd/poudriere/issues/178

> add option to nice builds

### [renice(8)](https://www.freebsd.org/cgi/man.cgi?query=renice&sektion=8&manpath=FreeBSD)

An example. Two (2) jobs allowed, one of two specified ports (`gpu-firmware-kmod`) building at the time of this screenshot: 

![2022-01-08 14-53-23](https://user-images.githubusercontent.com/192271/148651054-f188189b-baf0-4d04-a2c5-4de45b6b53cd.png)

Zooming in: 

![image](https://user-images.githubusercontent.com/192271/148651577-c2dad9d6-a4de-416e-8169-6dd9462c0b42.png)

#### Question

If I had used [htop(1)](https://www.freebsd.org/cgi/man.cgi?query=htop&sektion=1&manpath=FreeBSD-Ports) <key>F7</key> to increase the priority of the selected command (a parent) then the increase would take effect: 

* for the next port in the queue
* but not for the already running build of `gpu-firmware-kmod`

– true?

(Apologies for the basic question. I have used FreeBSD for years, but never really used `nice` or `renice`.)

----

#### Notes to self

<https://stackoverflow.com/a/1061699/38108>

> Children inherit the _current_ priority of a process when they're created. That means, if you renice the parent and start a child, it will have the modified priority. …

<https://stackoverflow.com/q/2463533/38108>

> python - Is it possible to renice a subprocess? - Stack Overflow

Via <https://stackoverflow.com/a/8165440/38108>: 

https://github.com/jedie/python-code-snippets/blob/master/CodeSnippets/reniceall.py

**Comment by grahamperrin** [comment]
Today I discovered [rtprio(1)](https://www.freebsd.org/cgi/man.cgi?query=rtprio&sektion=1&manpath=FreeBSD). 

[pkg and poudriere - a lightning trip - HackMD](https://hackmd.io/@dch/HkwIhv6x7) thanks @skunkwerks via <https://forums.freebsd.org/posts/563555>.

**Comment by ctyz1999** [comment]
It would really be nice if info from [https://forums.freebsd.org/posts/563555](https://forums.freebsd.org/posts/563555)
either comment 7 or 11 were included so all people could enjoy interactive responsiveness.
Could poudriere auto magically use those nice and idprio settings?
